src/pint/observatory/topo_obs.py

```diff
@@ -197,22 +218,97 @@ def __init__(
         if clock_fmt == "tempo" and clock_file == "time.dat" and tempo_code is None:
             raise ValueError("No tempo_code set for observatory '%s'" % name)
 
+        # GPS corrections
+        self.include_gps = include_gps
+
+        # BIPM corrections
+        # WARNING: `get_observatory` changes these after construction
+        self.include_bipm = include_bipm
+        self.bipm_version = bipm_version
         self.bogus_last_correction = bogus_last_correction
 
         self.tempo_code = tempo_code
+        self.itoa_code = itoa_code
+        if aliases is None:
+            aliases = []
+        for code in (tempo_code, itoa_code):
+            if code is not None:
+                aliases.append(code)
+
         self.origin = origin
+        super().__init__(
+            name,
+            aliases=aliases,
+            include_gps=include_gps,
+            include_bipm=include_bipm,
+            bipm_version=bipm_version,
+            overwrite=overwrite,
+        )
 
     def __repr__(self):
         aliases = [f"'{x}'" for x in self.aliases]
-        s = f"TopoObs('{self.name}' ({','.join(aliases)}) at [{self._loc_itrf.x}, {self._loc_itrf.y} {self._loc_itrf.z}]:\n{self.origin})"
+        s = f"TopoObs('{self.name}' ({','.join(aliases)}) at [{self.location.x}, {self.location.y} {self.location.z}]:\n{self.origin})"
         return s
 
     @property
     def timescale(self):
         return "utc"
 
+    @property
+    def as_dict(self):
```

Comment:
It would be good to have a round-trip test - ensure that sending something out to `as_dict` or `as_json` and then reading it back left everything unchanged. (Maybe Pyhont `==` is the right test?)

Reply:
I think that is done in `test_observatory`:
```
def test_json_observatory_output(sandbox):
    gbt_orig = get_observatory("gbt")
    load_observatories(io.StringIO(gbt_orig.as_json), overwrite=True)
    gbt_reload = get_observatory("gbt")

    for p in gbt_orig.__dict__:
        if not p in ["_clock"]:
            assert getattr(gbt_orig, p) == getattr(gbt_reload, p)
```
This tests round-trip through JSON.  I don't do a global `==` because some properties like `_clock` may be loaded from other tests.  So I check the properties separately.

```
def test_json_observatory_input_latlon(sandbox):
    gbt_orig = get_observatory("gbt")
    gbt_dict = gbt_orig.as_dict
    # remove ITRF
    del gbt_dict["gbt"]["itrf_xyz"]
    # add in geodetic
    gbt_dict["gbt"]["lat"] = gbt_orig.location.lat.value
    gbt_dict["gbt"]["lon"] = gbt_orig.location.lon.value
    gbt_dict["gbt"]["height"] = gbt_orig.location.height.value
    load_observatories(io.StringIO(json.dumps(gbt_dict)), overwrite=True)
    gbt_reload = get_observatory("gbt")

    for p in gbt_orig.__dict__:
        if not p in ["location", "_clock"]:
            # everything else should be identical
            assert getattr(gbt_orig, p) == getattr(gbt_reload, p)
    # check distance separately to allow for precision
    distance = gbt_orig.separation(gbt_reload)
    assert distance < 1 * u.m
```
Same here.  This dumps to `dict` and does the same thing.  Or did you have something else in mind.

Reply:
I was thinking a simple unmodified round-trip, which your first test does. Sorry I missed it!